config.json

```diff
@@ -0,0 +1,14 @@
+{
+  "tcr": {
+    "1": { "listId": 1, "contractAddress": "0x1854dae560abb0f399d8badca456663ca5c309d0" },
+    "4": { "contractAddress": "0xBb840456546496E7640DC09ba9fE06E67C157E1b" }
+  },
+  "dexPriceEstimator": {
+    "1": "https://dex-price-estimator.gnosis.io/",
+    "4": "https://dex-price-estimator.rinkeby.gnosis.io/"
+  },
+  "theGraphApi": {
+    "1": "https://api.thegraph.com/subgraphs/name/gnosis/protocol",
+    "4": "https://api.thegraph.com/subgraphs/name/gnosis/protocol-rinkeby"
+  }
+}
```

Comment:
I'll make a proposal with an example of a config. 
I think every piece of the config should fully optional, so if you create an empty yaml, or no yaml at all, would have sensible defaults.


```yaml
# Use a supported implementation with the default config
name: My cool web
logo:
  path: custom/my-logo.png
  width: 100
  height: 100

# Use a supported implementation with the default config
tokenList:
  type: 'json-file'

# Use a supported implementation with the default config
tokenList:
  type: 'gnosis-protocol-contract'


# Use a supported implementation with any other config
tokenList:
  type: 'json-file'
  config:
    path: custom/tokenList.json


# Example for the TCR you included in this PR
tcr:
  type: 'multi-tcr' # json-file, kleros, multi-tcr, none (for full list, it's not the default)
  config:    
    - networkId: 1
      listId: 1
      contractAddress: "0x1854dae560abb0f399d8badca456663ca5c309d0"
    - networkId: 4
      contractAddress: "0xBb840456546496E7640DC09ba9fE06E67C157E1b"


# Example of gas station config, although as any other config there should be a default (actually the following would be the default)
dexPriceEstimation:
  type: 'gnosis-gas-station'  
  config:
    defaultGas: standard # standard, lowest, safeLow, fast, fastest
    endpoints:
      - networkId: 1
        url: "https://dex-price-estimator.gnosis.io"
      - networkId: 4
        url: https://dex-price-estimator.rinkeby.gnosis.io

    
# Override any API implementation
tokenList:
  factory: custom/tokenList.ts
  config:
    constructorParamA: 1
    constructorParamB: 'Hi'


# All of the above configs can be used by the service/api factories to:
#   - Create an instance merging this config with the default config
#   - Normally it will be enough to provide just one implementation, ideally with some basic config parameters
#   - Ideally, we would allow to provide a "factory", that would be any custom implementation of the interface
```

Reply:
Yes, I 100% agree with the optional part, for the **custom** config.

Default app config though, I prefer to have some data there for two reasons:

1. separates code from configuration
2. helps to give concrete examples of what to change when customizing

Reply:
> ```
> # Use a supported implementation with the default config
> tokenList:
>   type: 'gnosis-protocol-contract'
> ```

And I don't get the `type` part on the config. How does that work?

Reply:
> Yes, I 100% agree with the optional part, for the custom config.

Not sure I follow what you like or dislikes :) 
I understand that you prefer if this doen't work by default with no-config.

My argument is that new version might bring new config, and to be backwards compatible we should have defaults.

The defaults can be just the `config-default.yaml` that get merged with `config.yaml`.
An user woud copy the default into `config.yaml` to start a configuration.

We could use YAML comments to document possible values

Reply:
> The defaults can be just the config-default.yaml that get merged with config.yaml.
An user woud copy the default into config.yaml to start a configuration.

Yes, that's what I mean :)

- custom user config: optional
- default app config: always present